api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricsServiceLifecycle.java

```diff
@@ -0,0 +1,241 @@
+            || "embedded_cass".equals(System.getProperty("hawkular-metrics.backend"))) {
+            long start = System.nanoTime();
+            while (state == State.STARTING
+                   // Give up after a minute. The deployment won't be failed and we'll continue to try to start the
+                   // service in the background.
+                   && NANOSECONDS.convert(1, MINUTES) > System.nanoTime() - start) {
+                Uninterruptibles.sleepUninterruptibly(1, SECONDS);
+            }
+        }
+    }
+
+    private void startMetricsService() {
+        if (state != State.STARTING) {
+            return;
+        }
+        LOG.info("Initializing metrics service");
+        connectionAttempts++;
+        try {
+            session = createSession();
+        } catch (Throwable t) {
+            Throwable rootCause = Throwables.getRootCause(t);
+            LOG.warn("Could not connect to Cassandra cluster - assuming its not up yet", rootCause);
+            // cycle between original and more wait time - avoid waiting huge amounts of time
+            long delay = 1 + ((connectionAttempts - 1) % 4);
+            LOG.warn("[{}] Retrying connecting to Cassandra cluster in [{}]s...", connectionAttempts, delay);
+            lifecycleExecutor.schedule(this::startMetricsService, delay, SECONDS);
+            return;
+        }
+        SchemaManager schemaManager = new SchemaManager(session);
+        try {
+            if (Boolean.parseBoolean(resetDb)) {
+                schemaManager.dropKeyspace(keyspace);
+            }
+            // This creates/updates the keyspace + tables if needed
+            schemaManager.createSchema(keyspace);
+            session.execute("USE " + keyspace);
+            LOG.info("Using a key space of '{}'", keyspace);
+            metricsService.startUp(session);
+            LOG.info("Metrics service started");
+            state = State.STARTED;
+        } catch (Throwable t) {
+            LOG.error("An error occurred trying to connect to the Cassandra cluster", t);
+            state = State.FAILED;
+        }
+    }
+
+    private Session createSession() {
+        Cluster.Builder clusterBuilder = new Cluster.Builder();
+        int port;
+        try {
+            port = Integer.parseInt(cqlPort);
+        } catch (NumberFormatException nfe) {
+            String defaultPort = CASSANDRA_CQL_PORT.defaultValue();
+            LOG.warn("Invalid CQL port '{}', not a number. Will use a default of {}", cqlPort, defaultPort);
+            port = Integer.parseInt(defaultPort);
+        }
+        clusterBuilder.withPort(port);
+        Arrays.stream(nodes.split(",")).forEach(clusterBuilder::addContactPoint);
+
+        Cluster cluster = null;
+        Session session = null;
+        try {
+            cluster = clusterBuilder.build();
+            session = cluster.connect("system");
+            return session;
+        } finally {
+            if (session == null && cluster != null) {
```

Comment:
Overall I like the changes for managing the Session outside of MetricsService in large part because I think that ultimately we should be sharing the same Session object across services. This is a good, necessary step towards making that possible.